app/validators/date-range.js

```diff
@@ -0,0 +1,7 @@
+import moment from 'moment'
+
+export default function validateDateRange(options) {
+  return (key, newValue) => {
+    return moment().range(options.min, options.max).contains(moment(newValue), false)
```

Comment:
The returned function should have access to the changes and model

```js
return (key, newValue, oldValue, changes, model) => {
  // This means we might need to assign the challenge to the contribution upon creation?
  // We currently hardcoded this in our backend..
  model.get('challenge.startsOn')
}
```

This might be overly specific for a "date-range" validator and might deserve its own validator, what do you think?